src/query/storages/parquet/src/parquet_rs/parquet_reader/reader.rs

```diff
@@ -165,22 +184,95 @@ impl crate::parquet_reader::ParquetReader for ParquetReader {
             metadata,
             column_chunks,
         };
-        let mut reader = row_group
-            .get_record_batch_reader(part.num_rows, selection)
+        let rows_to_read = match &selection {
+            None => part.num_rows,
+            Some(s) => s.row_count(),
+        };
+        let reader = row_group
+            .get_record_batch_reader(rows_to_read, selection)
             .unwrap();
-        let batch = reader.next().unwrap().map_err(|e| {
-            ErrorCode::BadBytes(format!(
-                "Cannot read parquet file, error: {:?}",
-                e.to_string()
-            ))
-        })?;
-        DataBlock::from_record_batch(&batch)
-            .map_err(|e| {
-                ErrorCode::BadBytes(format!(
-                    "Cannot convert record batch to data block, error: {:?}",
-                    e
-                ))
-            })
-            .map(|v| v.0)
+        Ok(Box::new(RowGroupDeserializer::new(
+            reader,
+            part.location.clone(),
+            part.num_rows,
+            part.num_rows,
+        )))
+    }
+}
+
+struct RowGroupDeserializer {
+    batch_size: usize,
+    num_rows_to_read: usize,
+    num_rows_readn: usize,
+    seq: usize,
+    location: String,
+    reader: ParquetRecordBatchReader,
+}
+
+impl RowGroupDeserializer {
+    fn new(
+        reader: ParquetRecordBatchReader,
```

Comment:
the reader already has a field named `batch_size`.

https://github.com/apache/arrow-rs/blob/5724cf21c23aa9d5a3ef06b6381cf267903746ee/parquet/src/arrow/arrow_reader/mod.rs#L318-L323

Reply:
this `batch_size` is only for debug in error message, like `location`

Reply:
the batch_size of reader is private

Reply:
ParquetRecordBatchReaderBuilder can build this reader.

Reply:
we already has the reader.